packages/frontend/src/components/FlowStepConfigurationModal/ChooseAppAndEvent/ChooseApp.tsx

```diff
@@ -45,12 +41,28 @@ interface ChooseAppProps {
   onSelectAppEvent: (app: IApp, triggerOrAction: ITrigger | IAction) => void
 }
 
+function checkIsAppDisabled(
+  appKey: string,
+  isAppSelectable: Record<string, boolean>,
+) {
+  if (Object.keys(isAppSelectable).includes(appKey)) {
+    return !isAppSelectable[appKey]
+  }
+
+  return false
+}
+
 export default function ChooseApp(props: ChooseAppProps) {
   const { apps, onSelectAppEvent } = props
   const launchDarkly = useContext(LaunchDarklyContext)
-  const { patchModalState, isTrigger, isLastStep } = useContext(
-    FlowStepConfigurationContext,
-  )
+  const { patchModalState, isTrigger, isLastStep, step, prevStepId } =
+    useContext(FlowStepConfigurationContext)
+
+  const isAppSelectable = useIsAppSelectable({
```

Comment:
isAppSelectable suggests a boolean
```suggestion
  const appSelectabilityMap = useIsAppSelectable({
```